tests/ops/integration_tests/test_privacy_request_logging.py

```diff
@@ -0,0 +1,162 @@
+            url,
+            headers=generate_auth_header(scopes=[PRIVACY_REQUEST_CREATE]),
+            json=[
+                {
+                    "policy_key": policy.key,
+                    "identity": {"email": provided_identity_value},
+                }
+            ],
+        )
+
+        privacy_request = response.json()["succeeded"][0]
+
+        extra = {
+            "system_key": None,
+            "connection_key": "zendesk_instance",
+            "action_type": "access",
+            "collection": "user",
+            "privacy_request_id": privacy_request["id"],
+            "method": "GET",
+            "url": "https://ethyca-test.zendesk.com/api/v2/users/search.json?query=test%40example.com",
+            "status_code": 401,
+            "error_group": ErrorGroup.authentication_error.value,
+        }
+
+        assert (
+            f"Connector request failed with status code 401. | {str(extra)}"
+            in loguru_caplog.text
+        )
+
+    @pytest.mark.usefixtures("typeform_runner")
+    async def test_erasure_error_logs(
+        self,
+        mock_send,
+        api_client,
+        url,
+        generate_auth_header,
+        erasure_policy,
+        loguru_caplog,
+        typeform_secrets,
+        provided_identity_value,
+    ):
+        masking_strict = CONFIG.execution.masking_strict
+        CONFIG.execution.masking_strict = False
+
+        response = api_client.post(
+            url,
+            headers=generate_auth_header(scopes=[PRIVACY_REQUEST_CREATE]),
+            json=[
+                {
+                    "policy_key": erasure_policy.key,
+                    "identity": {"email": provided_identity_value},
+                }
+            ],
+        )
+
+        privacy_request = response.json()["succeeded"][0]
+
+        extra = {
+            "system_key": None,
+            "connection_key": "typeform_instance",
+            "action_type": "erasure",
+            "collection": "user",
+            "privacy_request_id": privacy_request["id"],
+            "method": "DELETE",
+            "url": f"https://api.typeform.com/rtbf/{typeform_secrets['account_id']}/responses",
+            "body": '["test@example.com"]\n',
+            "status_code": 401,
+            "error_group": ErrorGroup.authentication_error.value,
+        }
+
+        assert (
+            f"Connector request failed with status code 401. | {str(extra)}"
+            in loguru_caplog.text
+        )
+
+        CONFIG.execution.masking_strict = masking_strict
+
+    @pytest.mark.usefixtures("klaviyo_runner")
+    async def test_consent_error_logs(
+        self,
+        mock_send,
+        klaviyo_runner,
+        api_client,
+        url,
+        generate_auth_header,
+        consent_policy,
+        provided_identity_and_consent_request,
+        loguru_caplog,
+        provided_identity_value,
+    ):
+        with pytest.raises(ClientUnsuccessfulException):
+            await klaviyo_runner.new_consent_request(
```

Comment:
I need to update this test to be triggered by API calls, the call to `new_consent_request` is just a placeholder for now.